diff --git a/backend/apps/rag/main.py b/backend/apps/rag/main.py
index 8631846e..2be96b6f 100644
--- a/backend/apps/rag/main.py
+++ b/backend/apps/rag/main.py
@@ -193,13 +193,14 @@ def update_embedding_model(
     update_model: bool = False,
 ):
     if embedding_model and app.state.config.RAG_EMBEDDING_ENGINE == "":
-        import sentence_transformers
+        import sentence_transformers, torch
 
-        app.state.sentence_transformer_ef = sentence_transformers.SentenceTransformer(
+        model = sentence_transformers.SentenceTransformer(
             get_model_path(embedding_model, update_model),
             device=DEVICE_TYPE,
             trust_remote_code=RAG_EMBEDDING_MODEL_TRUST_REMOTE_CODE,
         )
+        app.state.sentence_transformer_ef = torch.compile(model)
     else:
         app.state.sentence_transformer_ef = None
 
