diff --git a/backend/open_webui/utils/utils.py b/backend/open_webui/utils/utils.py
index cde953102..289856e98 100644
--- a/backend/open_webui/utils/utils.py
+++ b/backend/open_webui/utils/utils.py
@@ -2,7 +2,7 @@ import logging
 import uuid
 import jwt
 
-from datetime import UTC, datetime, timedelta
+from datetime import timezone, datetime, timedelta
 from typing import Optional, Union, List, Dict
 
 from open_webui.apps.webui.models.users import Users
@@ -42,7 +42,7 @@ def create_token(data: dict, expires_delta: Union[timedelta, None] = None) -> st
     payload = data.copy()
 
     if expires_delta:
-        expire = datetime.now(UTC) + expires_delta
+        expire = datetime.now(timezone.utc) + expires_delta
         payload.update({"exp": expire})
 
     encoded_jwt = jwt.encode(payload, SESSION_SECRET, algorithm=ALGORITHM)
